lib/nerves_livebook/redirect_nerves_local.ex

```diff
@@ -0,0 +1,30 @@
+defmodule NervesLivebook.RedirectNervesLocal do
+  @moduledoc false
+
+  @behaviour Plug
+
+  import Plug.Conn
+
+  @impl Plug
+  def init(opts) do
+    opts
+  end
+
+  @impl Plug
+  def call(conn, _opts) do
+    # Redirect nerves.local to the preferred hostname advertised over mDNS
+    if conn.host == "nerves.local" do
+      {:ok, hostname} = :inet.gethostname()
+      url = "http://#{hostname}.local"
```

Comment:
We should be able to preserve the pat/query like this:

```suggestion
      url =
        conn
        |> request_url()
        |> URI.parse()
        |> Map.put(:host, "#{hostname}.local")
        |> URI.to_string()
```

Reply:
Thanks! That works better.